src/tools/flash.c

```diff
@@ -208,13 +209,25 @@ int main(int ac, char** av)
     }
     else /* read */
     {
-        if ((o.addr >= sl->flash_base) && (o.size == 0) &&
-                (o.addr < sl->flash_base + sl->flash_size))
+        if(o.area == FLASH_OPTION_BYTES){
+            if(sl->chip_id == STLINK_CHIPID_STM32_F2){
+                uint32_t option_byte = 0;
+                err = stlink_read_option_bytes_f2(sl,&option_byte);
+                printf("%x\n",option_byte);
+            }else{
+                printf("This format is available for STM32F2 Only\n");
+            }
+        }
+        else if ((o.addr >= sl->flash_base) && (o.size == 0) &&
+                (o.addr < sl->flash_base + sl->flash_size)){
             o.size = sl->flash_size;
+            err = stlink_fread(sl, o.filename, o.format == FLASH_FORMAT_IHEX, o.addr, o.size);
```

Comment:
The call to stlink_fread to read out the content is placed inside the if intended to automatically fill in the correct flash/SRAM size.

Reply:
Thanks for comment. I will fixed today and submit pull request.
https://github.com/Orie22/stlink/commit/4a32157143547755cab260e58d1b00f8ddaa7780#diff-63af8289eae43567b379c0c480159f85R226